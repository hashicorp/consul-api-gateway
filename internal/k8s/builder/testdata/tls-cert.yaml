# Copyright (c) HashiCorp, Inc.
# SPDX-License-Identifier: MPL-2.0

apiVersion: api-gateway.consul.hashicorp.com/v1alpha1
kind: GatewayClassConfig
metadata:
  name: test-gateway-class-config
spec:
  consul:
    scheme: https
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: GatewayClass
metadata:
  name: test-gateway-class
spec:
  controller: "hashicorp.com/consul-api-gateway-gateway-controller"
  parametersRef:
    group: api-gateway.consul.hashicorp.com
    kind: GatewayClassConfig
    name: test-gateway-class-config
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: tls-cert-test
spec:
  gatewayClassName: test-gateway-class
  listeners:
  - protocol: HTTP
    port: 80
    name: http
    allowedRoutes:
      namespaces:
        from: Same
