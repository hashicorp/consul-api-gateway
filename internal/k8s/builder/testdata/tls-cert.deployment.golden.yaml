metadata:
  creationTimestamp: null
  labels:
    api-gateway.consul.hashicorp.com/created: "-62135596800"
    api-gateway.consul.hashicorp.com/managed: "true"
    api-gateway.consul.hashicorp.com/name: tls-cert-test
    api-gateway.consul.hashicorp.com/namespace: ""
  name: tls-cert-test
spec:
  replicas: 1
  selector:
    matchLabels:
      api-gateway.consul.hashicorp.com/created: "-62135596800"
      api-gateway.consul.hashicorp.com/managed: "true"
      api-gateway.consul.hashicorp.com/name: tls-cert-test
      api-gateway.consul.hashicorp.com/namespace: ""
  strategy: {}
  template:
    metadata:
      annotations:
        consul.hashicorp.com/connect-inject: "false"
      creationTimestamp: null
      labels:
        api-gateway.consul.hashicorp.com/created: "-62135596800"
        api-gateway.consul.hashicorp.com/managed: "true"
        api-gateway.consul.hashicorp.com/name: tls-cert-test
        api-gateway.consul.hashicorp.com/namespace: ""
    spec:
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - podAffinityTerm:
              labelSelector:
                matchLabels:
                  api-gateway.consul.hashicorp.com/created: "-62135596800"
                  api-gateway.consul.hashicorp.com/managed: "true"
                  api-gateway.consul.hashicorp.com/name: tls-cert-test
                  api-gateway.consul.hashicorp.com/namespace: ""
              topologyKey: kubernetes.io/hostname
            weight: 1
      containers:
      - args:
        - -log-json
        - -log-level
        - info
        - -gateway-host
        - $(IP)
        - -gateway-name
        - tls-cert-test
        - -gateway-namespace
        - test
        - -consul-http-address
        - $(HOST_IP)
        - -consul-http-port
        - "8500"
        - -consul-xds-port
        - "8502"
        - -envoy-bootstrap-path
        - /bootstrap/envoy.json
        - -envoy-sds-address
        - consul-api-gateway-controller.default.svc.cluster.local
        - -envoy-sds-port
        - "9090"
        command:
        - /bootstrap/consul-api-gateway
        - exec
        env:
        - name: IP
          valueFrom:
            fieldRef:
              fieldPath: status.podIP
        - name: HOST_IP
          valueFrom:
            fieldRef:
              fieldPath: status.hostIP
        - name: CONSUL_LOGIN_PARTITION
        - name: CONSUL_LOGIN_DATACENTER
        - name: CONSUL_DYNAMIC_SERVER_DISCOVERY
        - name: CONSUL_PARTITION
        - name: CONSUL_TLS_SERVER_NAME
        - name: PATH
          value: /:/sbin:/bin:/usr/bin:/usr/local/bin:/bootstrap
        - name: CONSUL_CACERT
          value: /consul/tls/ca.pem
        image: envoyproxy/envoy:v1.24-latest
        name: consul-api-gateway
        ports:
        - containerPort: 20000
          name: ready
          protocol: TCP
        - containerPort: 80
          name: http
          protocol: TCP
        readinessProbe:
          httpGet:
            path: /ready
            port: 20000
        resources: {}
        volumeMounts:
        - mountPath: /bootstrap
          name: bootstrap
        - mountPath: /certs
          name: certs
        - mountPath: /consul/tls
          name: ca
      initContainers:
      - command:
        - cp
        - /bin/discover
        - /bin/consul-api-gateway
        - /bootstrap/
        image: hashicorp/consul-api-gateway:0.5.3
        name: consul-api-gateway-init
        resources: {}
        volumeMounts:
        - mountPath: /bootstrap
          name: bootstrap
        - mountPath: /certs
          name: certs
        - mountPath: /consul/tls
          name: ca
      volumes:
      - emptyDir: {}
        name: bootstrap
      - emptyDir: {}
        name: certs
      - name: ca
        secret:
          items:
          - key: consul-ca-cert
            path: ca.pem
          optional: false
          secretName: tls-cert-test
status: {}
